### DISCUSSION TITLE
Can I use auth, axios, proxy, i18n with nuxt 3?

### DISCUSSION THREAD

**Discussion by n4an** [nuxt/nuxt#18186] [General]
Can I use auth, axios, proxy, i18n with nuxt 3?

**Comment by kissu** [comment]
Probably. What did you tried so far? Do you face any issues that you can share us in a repro? 😁

**Reply by n4an** [reply]
nothing works anymore. with vue-class-component, vue-property-decorator

**Reply by kissu** [reply]
Do you have a repro? Hard to help without concrete code to look at.
Also, do remember that Nuxt3 is still in beta so don't expect it to be 100% production-ready.

**Reply by n4an** [reply]
> Do you have a repro? Hard to help without concrete code to look at. Also, do remember that Nuxt3 is still in beta so don't expect it to be 100% production-ready.

it is not need a repo. simply install nuxt 3 and i18n, axios and auth module. And will see that nothing not work

**Reply by n4an** [reply]
First and so on

Cannot start nuxt:  Cannot read property 'resolveAlias' of undefined                                                                                                         13:59:12

  at Object._default (node_modules/@nuxtjs/i18n/src/index.js:34:42)
  at installModule (node_modules/@nuxt/kit/dist/index.mjs:1196:17)
  at initNuxt (node_modules/nuxt3/dist/index.mjs:817:11)
  at async load (node_modules/nuxi/dist/chunks/dev.mjs:6762:9)
  at async Object.invoke (node_modules/nuxi/dist/chunks/dev.mjs:6793:5)
  at async _main (node_modules/nuxi/dist/chunks/index.mjs:417:7)

**Comment by n4an** [comment]
ERROR  Error compiling template:  {                                                                                                                                                 14:13:27
  src: '/var/www/html/nuxt3-app/node_modules/@nuxtjs/axios/lib/plugin.js',
  fileName: 'axios.js',
  options: {
    baseURL: 'http://localhost:3000/',
    browserBaseURL: 'http://localhost:3000/',
    credentials: true,
    debug: false,
    progress: true,
    proxyHeaders: true,
    proxyHeadersIgnore: [
      'accept',
      'cf-connecting-ip',
      'cf-ray',
      'content-length',
      'content-md5',
      'content-type',
      'host',
      'x-forwarded-host',
      'x-forwarded-port',
      'x-forwarded-proto'
    ],
    proxy: false,
    retry: false,
    https: false,
    headers: {
      common: [Object],
      delete: {},
      get: {},
      head: {},
      post: {},
      put: {},
      patch: {}
    },
    globalName: 'nuxt'
  },
  filename: 'axios.js',
  dst: '/var/www/html/nuxt3-app/.nuxt/axios.js'
}


 ERROR  Error compiling template:  {                                                                                                                                                 14:13:27
  src: '/var/www/html/nuxt3-app/node_modules/@nuxtjs/auth/lib/module/plugin.js',
  fileName: 'auth/plugin.js',
  options: {
    options: {
      resetOnError: false,
      scopeKey: 'scope',
      rewriteRedirects: true,
      fullPathRedirect: false,
      watchLoggedIn: true,
      redirect: [Object],
      vuex: [Object],
      cookie: [Object],
      localStorage: false,
      token: [Object],
      refresh_token: [Object],
      defaultStrategy: 'local'
    },
    strategies: [
      [Object]
    ],
    uniqueSchemes: [
      './schemes/local.js'
    ],
    strategyScheme: Map(1) {
      [Object] => './schemes/local.js'
    }
  },
  write: true,
  filename: 'auth/plugin.js',
  dst: '/var/www/html/nuxt3-app/.nuxt/auth/plugin.js'
}

**Comment by imfaisalkh** [comment]
I agree, I've tried nuxt axios and i18n modules and they does not seems to work with nuxt 3.

**Reply by r9mp** [reply]
You can follow this topic if you want to see the current status of your modules : nuxt/nuxt#16725
Currently, only i18n is reported in it.

**Comment by xiaobai-world** [comment]
on @nuxt/axios, i got problem with  "Cannot start nuxt:  options is not defined "
![image](https://user-images.githubusercontent.com/82394937/137594192-9ee01d38-f8fe-42aa-afd5-6af33e994cb5.png)

**Comment by kevhof** [comment]
As there is only serverMiddlewares and no more client middlewares in Nuxt 3 at the moment, I would say that nuxt auth can't work no.
